### PR TITLE
Nukeops hud

### PR DESCRIPTION
**Pull request by IntegerTempest**
## About the PR
damn hud so that your fucking friends wouldn't shoot at you
in a separate PR I plan to make a night vision when will this PR be accepted https://github.com/space-wizards/space-station-14/pull/17114

## Why / Balance
added as opposed to sec hud

## Media
https://github.com/space-wizards/space-station-14/assets/30300017/499a9915-b8c5-434a-aef1-68b1e1fa8e67

**Changelog**
:cl:
- add: Added syndicate hud and syndicate visor for nuclear operatives

### CODE REVIEW

**Comment by IntegerTempest** [comment]
in a separate PR I plan to make a night vision

**Comment by GoodWheatley** [comment]
This should be an innate nuke ops ability instead of tying it to a visor

**Comment by IntegerTempest** [comment]
> This should be an innate nuke ops ability instead of tying it to a visor

I think this is not logical, I absolutely do not like the fact that they will be able to see their allies without any devices, implants, etc.

**Comment by notquitehadouken** [comment]
i think this should be a visor because it gives nukeops another thing to lose (lol) and matches more with the sechud in "Seeing who people are" than wonder vision that nukies magically have or implants, which are usually used for special on-command abilities instead of permanent sechud

**Comment by notquitehadouken** [comment]
> This is a nice counterpart to the sechud. Syndicate operatives (or nukies in this case) won't need sunglasses or the like anyways, their masks have innate protection against welding and flashes!

i've spent my entire life thinking they only protected against flashing
~~also if you're in a situation where you need to use a welder as nukeop then _something has gone wrong_~~

**Comment by notquitehadouken** [comment]
> it should just scan for syndicate access on ID

what about commander and agent and how they get cap id 2 seconds in because they don't have a regular agent id
instead nukeops should have a funny implant that will show up on the visors

**Comment by Ilya246** [comment]
> > it should just scan for syndicate access on ID
> 
> what about commander and agent and how they get cap id 2 seconds in because they don't have a regular agent id instead nukeops should have a funny implant that will show up on the visors

make their IDs be agent IDs too, maybe just make them parent off of it

**Comment by notquitehadouken** [comment]
> > > it should just scan for syndicate access on ID
> > 
> > 
> > what about commander and agent and how they get cap id 2 seconds in because they don't have a regular agent id instead nukeops should have a funny implant that will show up on the visors
> 
> make their IDs be agent IDs too, maybe just make them parent off of it

shhhhhhhhhhhhhhhhhhhhhhh

**Comment by JoeHammad1844** [comment]
Having it be innate (or even better, an implant) is a lot better for future proofing
1. Leaves the eyes free for night vision whenever that's made
2. In ss13 nukie guns have Syndicate firing pins, which means they only fire for someone with a Syndicate implant, which would be great (the firing pins can also be taken out and replaced, but science can only make mindshield firing pins, which only fire for people with mindshields)

**Comment by JoeHammad1844** [comment]
It shouldn't be sechud though, too much eye clutter and nukies just don't need it, they don't need to see what ID the captain or whoever else is shooting them has

**Comment by notquitehadouken** [comment]
> It shouldn't be sechud though, too much eye clutter and nukies just don't need it, they don't need to see what ID the captain or whoever else is shooting them has

it would probably help them identify where cap is if they aren't wearing bright blue constantly and they don't have time to perform trigonometry with their pinpointer

**Comment by LordEclipse** [comment]
> Having it be innate (or even better, an implant) is a lot better for future proofing
> 
>     1. Leaves the eyes free for night vision whenever that's made
> 
>     2. In ss13 nukie guns have Syndicate firing pins, which means they only fire for someone with a Syndicate implant, which would be great (the firing pins can also be taken out and replaced, but science can only make mindshield firing pins, which only fire for people with mindshields)

Firing pins suck, let people use guns

**Comment by chromiumboy** [comment]
IMO, nukies should just innately know who their fellow nukies are, just like revs do. You could use the exact dame system

**Comment by JoeHammad1844** [comment]
> Firing pins suck, let people use guns

Could you be more constructive

**Comment by JoeHammad1844** [comment]
But let's not get too deep into firing pins because that's off topic

**Comment by JoeHammad1844** [comment]
> IMO, nukies should just innately know who their fellow nukies are, just like revs do. You could use the exact same system

You get trained by the gorlex marauders for 5 years, you and your squad are like brothers, you eat,sleep and cry together, you're like a close family and you completely forget who's who when you take your deal with it glasses off

**Comment by JoeHammad1844** [comment]
> i think this should be a visor because it gives nukeops another thing to lose (lol)

another reason to put your microbomb in? based

(seeing allies should be innate but not sechud)

**Comment by Emisse** [comment]
I am fine with this for now, in the future if it should be innate we can just do that.

File: Content.Shared/Overlays/ShowSyndicateIconsComponent.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Overlays;
+
+/// <summary>
+/// 
+/// </summary>
+[RegisterComponent, NetworkedComponent]
```
Comment (metalgearsloth) [inline Content.Shared/Overlays/ShowSyndicateIconsComponent.cs:8]: ?

File: Content.Shared/NukeOps/NukeOperativeComponent.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    /// <summary>
+    ///     Path to antagonist alert sound.
+    /// </summary>
+    [DataField("greetSoundNotification")]
+    public SoundSpecifier GreetSoundNotification = new SoundPathSpecifier("/Audio/Ambience/Antag/nukeops_start.ogg");
+
+    /// <summary>
+    ///     
+    /// </summary>
+    [DataField("syndStatusIcon", customTypeSerializer: typeof(PrototypeIdSerializer<StatusIconPrototype>))]
```
Comment (metalgearsloth) [inline Content.Shared/NukeOps/NukeOperativeComponent.cs:25]: ?

**Pull request merged**